File: leetcode/golang/2739_total_distance_traveled.go
// https://leetcode.cn/problems/total-distance-traveled

func distanceTraveled(mainTank int, additionalTank int) int {
    res, count := 0, 0
    for mainTank != 0 {
        res += 10
        count++
        mainTank--
        if count % 5 == 0 && additionalTank > 0{
            mainTank++
            additionalTank--
        }
    }
    return res
}

// 模拟，时间复杂度为O(n)

File: leetcode/c++/2739_total_distance_traveled.cpp
// https://leetcode.cn/problems/total-distance-traveled

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int distanceTraveled(int mainTank, int additionalTank) {
        int res = 0, flag = 0;
        while (mainTank) {
            res += 10;
            flag++;
            mainTank--;
            if (flag % 5 == 0 && additionalTank) {
                mainTank++;
                additionalTank--;
            }
        }
        return res;
    }
};

// 模拟，时间复杂度为O(n)